### ISSUE TITLE
Support python 3

### ISSUE THREAD

**Issue by ThiefMaster** [ThiefMaster/maildump#7]

**Comment by pydanny** [comment]
Alas, the lack of Python 3 support is causing problems with cookiecutter-django. Is there any chance this will support happen soon? 

Otherwise we have to remove maildump from cookiecutter-django. Reference https://github.com/pydanny/cookiecutter-django/issues/297

**Comment by wldcordeiro** [comment]
It appear that some of the dependencies also lack Python 3 support so the task of adding Python 3 isn't quite so straightforward.

So far in my attempt I've determined that `gevent-socketio` doesn't support 3, I don't know about others yet since it's the first to error out.

**Comment by wldcordeiro** [comment]
I updated the requirements.txt to point to specific versions of dependencies. Here's the list and whether they support Python 3.

```
Flask==0.10.1  # Py3 Ready
Flask-Assets==0.10  # Py3 Ready
Logbook==0.10.1  # Py3 Ready
beautifulsoup4==4.4.0  # Py3 Ready
cssmin==0.2.0  # Not sure from docs
cssprefixer==1.4.1  # Not sure from docs
cssutils==1.0  # Py3 Ready
gevent==1.1b2  # Py3 Ready
gevent-socketio==0.3.6  # Not ready
gevent-websocket==0.9.5  # Not ready
html5lib==0.999999  # Py3 Ready
lockfile==0.10.2  # Py3 Ready
passlib==1.6.5  # Py3 Ready
pyScss==1.3.4  # Py3 Ready
python-daemon==2.0.5  # Py3 Ready
pytz==2015.4  # Py3 Ready
six==1.9.0  # Used for Py3 support
```

So it looks like 2-4 dependencies need to be updated as well.

**Comment by wldcordeiro** [comment]
Tested cssmin, works with Python 3.

**Comment by wldcordeiro** [comment]
cssprefixer doesn't work in Python 3.

**Comment by wldcordeiro** [comment]
I've gotten to the point where I'm able to run the `run.py` script but currently get errors with the CSS. I think this is probably a result of the cssprefixer having issues. I had to fork gevent-socketio to get to this point. Current traceback included.

```
Traceback (most recent call last):
  File "/home/wldcordeiro/.pyenv/versions/maildump/lib/python3.4/site-packages/flask/app.py", line 1817, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/wldcordeiro/.pyenv/versions/maildump/lib/python3.4/site-packages/flask/app.py", line 1477, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/wldcordeiro/.pyenv/versions/maildump/lib/python3.4/site-packages/flask/app.py", line 1381, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/wldcordeiro/.pyenv/versions/maildump/lib/python3.4/site-packages/flask/_compat.py", line 33, in reraise
    raise value
  File "/home/wldcordeiro/.pyenv/versions/maildump/lib/python3.4/site-packages/flask/app.py", line 1475, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/wldcordeiro/.pyenv/versions/maildump/lib/python3.4/site-packages/flask/app.py", line 1461, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/wldcordeiro/Documents/github-projects/maildump/maildump/web.py", line 66, in home
    return render_template('index.html', version=get_version())
  File "/home/wldcordeiro/.pyenv/versions/maildump/lib/python3.4/site-packages/flask/templating.py", line 128, in render_template
    context, ctx.app)
  File "/home/wldcordeiro/.pyenv/versions/maildump/lib/python3.4/site-packages/flask/templating.py", line 110, in _render
    rv = template.render(context)
  File "/home/wldcordeiro/.pyenv/versions/maildump/lib/python3.4/site-packages/jinja2/environment.py", line 989, in render
    return self.environment.handle_exception(exc_info, True)
  File "/home/wldcordeiro/.pyenv/versions/maildump/lib/python3.4/site-packages/jinja2/environment.py", line 754, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/wldcordeiro/.pyenv/versions/maildump/lib/python3.4/site-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "/home/wldcordeiro/Documents/github-projects/maildump/maildump/templates/index.html", line 10, in top-level template code
    {% assets 'css_all' -%}
  File "/home/wldcordeiro/.pyenv/versions/maildump/lib/python3.4/site-packages/webassets/ext/jinja2.py", line 186, in _render_assets
    urls = bundle.urls()
  File "/home/wldcordeiro/.pyenv/versions/maildump/lib/python3.4/site-packages/webassets/bundle.py", line 787, in urls
    urls.extend(bundle._urls(new_ctx, extra_filters, *args, **kwargs))
  File "/home/wldcordeiro/.pyenv/versions/maildump/lib/python3.4/site-packages/webassets/bundle.py", line 747, in _urls
    return [self._make_output_url(ctx)]
  File "/home/wldcordeiro/.pyenv/versions/maildump/lib/python3.4/site-packages/webassets/bundle.py", line 704, in _make_output_url
    version = self.get_version(ctx, refresh=ctx.auto_build)
  File "/home/wldcordeiro/.pyenv/versions/maildump/lib/python3.4/site-packages/webassets/bundle.py", line 315, in get_version
    'determined dynamically, because: %s') % (self, reason))
webassets.exceptions.BundleError: Cannot find version of <Bundle output=assets/bundle.%(version)s.css, filters=[<webassets.filter.cssrewrite.CSSRewrite object at 0x7f12fe8e67f0>, <maildump.util.CSSPrefixer object at 0x7f12fe8da710>, <webassets.filter.cssmin.CSSMin object at 0x7f12fe8e6828>], contents=('css/reset.css', 'css/jquery-ui.css', <Bundle output=assets/maildump.%(version)s.css, filters=[<webassets.filter.pyscss.PyScss object at 0x7f12fe8e6668>], contents=('css/maildump.scss',)>)>. There is no manifest which knows the version, and it cannot be determined dynamically, because: output target has a placeholder```

This makes me question though, why we even need cssmin, cssprefixer, cssutils, html5lib or pyScss, these kinds of things could be offloaded to some JS with something like Grunt or Gulp and reduce the dependency count for this project.

**Comment by ThiefMaster** [comment]
html5lib is used by beautifulsoup to parse html emails: https://github.com/ThiefMaster/maildump/blob/master/maildump/web.py#L163

I don't really like the idea of having to run separate steps to build assets, especially during development.

**Comment by wldcordeiro** [comment]
@ThiefMaster Ah okay, I wasn't sure what html5lib was being used for. If this tool is meant mainly for development, it seems like the assets aren't a priority and could be packaged precompiled with the sources included if anyone wants to customize it for their own use rather than having additional dependencies to build the assets.

**Comment by rafspiny** [comment]
@ThiefMaster I planned to use `maildumper` to test certain functionalities I am developing where I work. I really wanted to use it with python3, so I made a fork and played with it a little bit. I would like your opinion on it.

One open question:
- gevent-socketio latest release is not py3 ready, but its latest committed version (2016) is py3 ready. Thus, I specified to install the git version for python_version >= 3.4 in requirements.txt

I got ride of `cssprefixer`. I also tried to use `AutoPrefixer` but I did not like the dependency on node
